modules/nlu/src/backend/application/training-repo.ts

```diff
@@ -46,29 +35,69 @@ export class TrainingRepository implements TrainingRepository {
   public get = async (trainId: TrainingId): Promise<TrainingSession | undefined> => {
     const { botId, language } = trainId
 
-    return this._database
-      .from(TABLE_NAME)
+    return this.table
       .where({ botId, language })
       .select('*')
       .first()
   }
 
+  public getAll = async (): Promise<TrainingSession[]> => {
+    return this.table.select('*')
+  }
+
   public query = async (query: Partial<TrainingSession>): Promise<TrainingSession[]> => {
-    return this._database
-      .from(TABLE_NAME)
-      .where(query)
-      .select('*')
+    return this.table.where(query).select('*')
   }
 
   public delete = async (trainId: TrainingId): Promise<void> => {
     const { botId, language } = trainId
-    return this._database
-      .from(TABLE_NAME)
-      .where({ botId, language })
-      .delete()
+    return this.table.where({ botId, language }).delete()
   }
 
   public clear = async (): Promise<void[]> => {
-    return this._database.from(TABLE_NAME).delete('*')
+    return this.table.delete('*')
+  }
+}
+
+export type ITrainingRepository = I<TrainingRepository>
+export class TrainingRepository extends TransactionContext implements TrainingRepository {
```

Comment:
We've talked about this, but just make sure a **set** and **delete** operation is only doable from inside a transaction

Reply:
Done :D